Mathlib/LinearAlgebra/Matrix/SesquilinearForm.lean

```diff
@@ -80,196 +89,205 @@ section AuxToMatrix
+end CommSemiring
 
 end AuxToMatrix
 
 section ToMatrix'
 
-/-! ### Bilinear forms over `n → R`
+/-! ### Bilinear maps over `n → R`
 
-This section deals with the conversion between matrices and sesquilinear forms on `n → R`.
+This section deals with the conversion between matrices and sesquilinear maps on `n → R`.
 -/
 
-
-variable [CommSemiring R] [Semiring R₁] [Semiring R₂]
+variable [CommSemiring R] [AddCommMonoid N₂] [Module R N₂] [Semiring R₁] [Semiring R₂]
+  [SMulCommClass R R N₂] [Semiring S₁] [Semiring S₂] [Module S₁ N₂] [Module S₂ N₂]
+  [SMulCommClass S₁ R N₂] [SMulCommClass S₂ R N₂] [SMulCommClass S₂ S₁ N₂]
+variable {σ₁ : R₁ →+* S₁} {σ₂ : R₂ →+* S₂}
 variable [Fintype n] [Fintype m]
 variable [DecidableEq n] [DecidableEq m]
-variable {σ₁ : R₁ →+* R} {σ₂ : R₂ →+* R}
 
-/-- The linear equivalence between sesquilinear forms and `n × m` matrices -/
-def LinearMap.toMatrixₛₗ₂' : ((n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] R) ≃ₗ[R] Matrix n m R :=
-  {
-    LinearMap.toMatrix₂Aux (fun i => stdBasis R₁ (fun _ => R₁) i 1) fun j =>
+/-- The linear equivalence between sesquilinear maps and `n × m` matrices -/
+def LinearMap.toMatrixₛₗ₂' : ((n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] N₂) ≃ₗ[R] Matrix n m N₂ :=
+  { LinearMap.toMatrix₂Aux (R := R) (fun i => stdBasis R₁ (fun _ => R₁) i 1) fun j =>
       stdBasis R₂ (fun _ => R₂) j
         1 with
-    toFun := LinearMap.toMatrix₂Aux _ _
+    toFun := LinearMap.toMatrix₂Aux (R := R) _ _
     invFun := Matrix.toLinearMap₂'Aux σ₁ σ₂
-    left_inv := LinearMap.toLinearMap₂'Aux_toMatrix₂Aux
-    right_inv := Matrix.toMatrix₂Aux_toLinearMap₂'Aux }
+    left_inv := LinearMap.toLinearMap₂'Aux_toMatrix₂Aux (R := R)
+    right_inv := Matrix.toMatrix₂Aux_toLinearMap₂'Aux (R := R) }
 #align linear_map.to_matrixₛₗ₂' LinearMap.toMatrixₛₗ₂'
 
-/-- The linear equivalence between bilinear forms and `n × m` matrices -/
-def LinearMap.toMatrix₂' : ((n → R) →ₗ[R] (m → R) →ₗ[R] R) ≃ₗ[R] Matrix n m R :=
+/-- The linear equivalence between bilinear maps and `n × m` matrices -/
+def LinearMap.toMatrix₂' : ((n → S₁) →ₗ[S₁] (m → S₂) →ₗ[S₂] N₂) ≃ₗ[R] Matrix n m N₂ :=
   LinearMap.toMatrixₛₗ₂'
 #align linear_map.to_matrix₂' LinearMap.toMatrix₂'
 
 variable (σ₁ σ₂)
 
-/-- The linear equivalence between `n × n` matrices and sesquilinear forms on `n → R` -/
-def Matrix.toLinearMapₛₗ₂' : Matrix n m R ≃ₗ[R] (n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] R :=
+/-- The linear equivalence between `n × n` matrices and sesquilinear maps on `n → R` -/
+def Matrix.toLinearMapₛₗ₂' : Matrix n m N₂ ≃ₗ[R] (n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] N₂ :=
   LinearMap.toMatrixₛₗ₂'.symm
 #align matrix.to_linear_mapₛₗ₂' Matrix.toLinearMapₛₗ₂'
 
-/-- The linear equivalence between `n × n` matrices and bilinear forms on `n → R` -/
-def Matrix.toLinearMap₂' : Matrix n m R ≃ₗ[R] (n → R) →ₗ[R] (m → R) →ₗ[R] R :=
+/-- The linear equivalence between `n × n` matrices and bilinear maps on `n → R` -/
+def Matrix.toLinearMap₂' : Matrix n m N₂ ≃ₗ[R] (n → S₁) →ₗ[S₁] (m → S₂) →ₗ[S₂] N₂ :=
```

Comment:
Same comment about the `R` argument being explicit here.

Reply:
Presumably the same applies to `Matrix.toLinearMapₛₗ₂'` and `LinearMap.toMatrixₛₗ₂'`?